client/src/pages/locations/Form.js

```diff
@@ -110,6 +125,7 @@ const LocationForm = ({ edit, title, initialValues }) => {
         const marker = {
           id: values.uuid || 0,
           name: _escape(values.name) || "", // escape HTML in location name!
+          type: _escape(values.type) || "", // escape HTML in location type!
```

Comment:
That line is essential because when we remove it leaflet marker breaks the functionality of location type drop-down. I can change/remove the comment but I suggest we keep those lines.